src/index.js

```diff
@@ -74,28 +78,59 @@ export const documentToFlowTypes = (
         .map((item) => {
             if (item.kind === 'FragmentDefinition') {
                 const name = item.name.value;
+                const kids = {};
                 const code = `export type ${name} = ${generateFragmentType(
                     schema,
                     item,
-                    config,
+                    {
+                        ...config,
+                        path: [name],
+                        allObjectTypes: options?.exportAllObjectTypes
+                            ? kids
+                            : null,
+                    },
                 )};`;
-                return {name, typeName: name, code, isFragment: true};
+                const extraTypes: {[key: string]: string} = {};
+                Object.keys(kids).forEach((k) => {
+                    // eslint-disable-next-line flowtype-errors/uncovered
+                    extraTypes[k] = generate(kids[k]).code;
+                });
+                return {
+                    name,
+                    typeName: name,
+                    code,
+                    isFragment: true,
+                    extraTypes,
+                };
             }
             if (
                 item.kind === 'OperationDefinition' &&
                 (item.operation === 'query' || item.operation === 'mutation') &&
                 item.name
             ) {
+                const kids = {};
                 const name = item.name.value;
-                const response = generateResponseType(schema, item, config);
-                const variables = generateVariablesType(schema, item, config);
+                const response = generateResponseType(schema, item, {
+                    ...config,
+                    path: [name],
+                    allObjectTypes: options?.exportAllObjectTypes ? kids : null,
+                });
+                const variables = generateVariablesType(schema, item, {
+                    ...config,
+                    path: [name],
+                });
```

Comment:
Is it necessary to pass different customizations of `config` to each of these functions?